spec/stubs/output_handler_stub.rb

```diff
@@ -0,0 +1,11 @@
+class OutputHandlerStub
+  attr_accessor :output
+
+  def initialize
+    self.output = ""
+  end
+
+  def print(string)
+    self.output = self.output + "\n" + string
+  end
+end
```

Comment:
I may take back what I said about premature abstraction. I do prefer that the tests can rely on something like this instead of [directly mocking library methods](https://www.markhneedham.com/blog/2009/12/13/tdd-only-mock-types-you-own/?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed%3A+MarkNeedham+%28Mark+Needham%29).